lib/graphql/schema/field.rb

```diff
@@ -233,14 +233,13 @@ def initialize(type: nil, name: nil, owner: nil, null: true, description: :not_g
           end
         end
 
-        # TODO: I think non-string/symbol hash keys are wrongly normalized (eg `1` will not work)
-        method_name = method || hash_key || name_s
+        @hash_key = hash_key
         @dig_keys = dig
-        resolver_method ||= name_s.to_sym
+        @object_method = method
+        @resolver_method = (resolver_method || hash_key || method || name_s).to_sym
+        @method_sym = @resolver_method
+        @method_str = @resolver_method.to_s
```

Comment:
This should continue using `String#-@` ([doc](https://ruby-doc.org/core-3.0.2/String.html#method-i-2D-40)) -- that might improve the retained memory in the benchmark, too. 

```suggestion
        @method_str = -@resolver_method.to_s
```

Reply:
f8a7a8552